src/parser.js

```diff
@@ -0,0 +1,112 @@
+import cheerio from 'cheerio'
+import axios from 'axios'
+
+async function parseFromSOCURL(SOCURL) {
+  // Request the data from the SOC URL
+  const res = await axios.get(SOCURL).catch(err => console.error(err))
+  if (res == null || (res.status < 200 || res.status >= 300)) {
+    console.error(`An error occurred trying to get the data from the URL ${SOCURL}`)
+    console.error(res)
+  }
+  const { data } = res
+
+  // Parse the HTML
+  const $ = cheerio.load(data)
+
+  // Class ID & internal ID
+  const classID = $('div').attr('id')?.match(/\w+/)[0]
+  const internalIDs = $(`div#${classID}-children > div`).toArray().map(e => $(e).attr('id'))
+
+  // Get the info for each lecture
+  const elems = $(`div#${classID}-children > div`).toArray()
+  const parsePromises = elems.map(async (e, index) => parseElem($, e, internalIDs[index]))
+  const classesInfo = await Promise.all(parsePromises)
+
+  return { classID, internalIDs, classesInfo }
+}
+
+/**
+ * Example: Select Computer Science (COM SCI) 31 - Introduction to Computer Science I Lec 1
+ * We ignore the "Select" part of this
+ * Group 1 - Match the department name - Computer Science
+ * Group 2 - Match the short departent name - COM SCI
+ * Group 3 - Match the course number in series - 31
+ * Group 4 - Match the course description - Introduction to Computer Science I
+ * Group 5 - Match the lecture section name
+ */
+const COURSE_NAME_REGEX =
+  /Select ([\w\s]+) \(([\w\s]+)\)\s+([\w\d]+)\s+-\s+([\w\s]+) ((Lec|Lab) (\d+))/
```

Comment:
Decent, good attempt. Passable. Reasonable. Fine

Lol I have no way to review this, but I believe in you :)